### PR TITLE
Added styles to footer view

### PR DESCRIPTION
**Pull request by vinothvino42**
Issue: #564 

In this PR, I added styles for the footer view of the page. You can apply all box layout styles like padding, margin, borderRadius, backgroundGradient, etc to the footer

### CODE REVIEW

**Comment by amin-nas** [comment]
@vinothvino42 @vusters please wrap this up and merge it as we need it for customer apps.

File: lib/framework/view/page.dart [hunk 1]
```diff
@@ -554,18 +556,30 @@ class PageState extends State<Page> {
   Widget? _buildFooter(ScopeManager scopeManager, SinglePageModel pageModel) {
     // Footer can only take 1 child by our design. Ignore the rest
     if (pageModel.footer != null && pageModel.footer!.children.isNotEmpty) {
+      final footerStyles = pageModel.footer?.styles;
+      final boxController = BoxController()
+        ..padding = Utils.getInsets(footerStyles?['padding'],
+            fallback:
```
Comment (vusters) [inline lib/framework/view/page.dart:562]: why need hardcoded fallback?

File: lib/framework/view/page.dart [hunk 2]
```diff
@@ -554,18 +556,30 @@ class PageState extends State<Page> {
...
+                const EdgeInsets.only(left: 16, right: 16, top: 16, bottom: 32))
+        ..margin = Utils.optionalInsets(footerStyles?['margin'])
+        ..width = Utils.optionalInt(footerStyles?['width'])
+        ..height = Utils.getInt(footerStyles?['height'], fallback: 110)
```
Comment (vusters) [inline lib/framework/view/page.dart:566]: remove all the fallbacks. It'll work still right?

**Review by vusters — APPROVED** [review]
Please see the comments. Otherwise good to go

File: lib/framework/view/page.dart (hunk 1 above)
Comment (vinothvino42) [reply lib/framework/view/page.dart]: Removed it

File: lib/framework/view/page.dart (hunk 2 above)
Comment (vinothvino42) [reply lib/framework/view/page.dart]: Yes, It'll work. I removed it

**Pull request merged**